File: packages/rag/src/api/evaluation.ts
/**
 * RAG API - Evaluation
 * 质量评估 API
 */
import { fetchApi } from '@packages/core/api/core';

export interface GoldenSampleCreate {
  kb_id: string;
  question: string;
  expected_answer: string;
  expected_context_ids?: string[];
  metadata?: Record<string, any>;
}

export interface GoldenSampleResponse {
  id: string;
  kb_id: string;
  question: string;
  expected_answer: string;
  expected_context_ids?: string[];
  metadata?: Record<string, any>;
}

export interface EvaluationRunCreate {
  kb_id: string;
  name?: string;
  sample_ids?: string[];
  metrics?: string[];
}

export interface EvaluationRunResponse {
  id: string;
  kb_id: string;
  name?: string;
  status: string;
  results?: any[];
  created_at: string;
  completed_at?: string;
}

export const createGoldenSample = async (data: GoldenSampleCreate) => {
  return fetchApi<GoldenSampleResponse>('/api/v1/evaluation/golden-samples', {
    method: 'POST',
    body: JSON.stringify(data),
  });
};

export const listGoldenSamples = async (kb_id?: string, limit?: number) => {
  const params = new URLSearchParams();
  if (kb_id) params.set('kb_id', kb_id);
  if (limit) params.set('limit', String(limit));
  const qs = params.toString();
  return fetchApi<GoldenSampleResponse[]>(`/api/v1/evaluation/golden-samples${qs ? `?${qs}` : ''}`);
};

export const deleteGoldenSample = async (sample_id: string) => {
  return fetchApi<void>(`/api/v1/evaluation/golden-samples/${sample_id}`, { method: 'DELETE' });
};

export const createEvaluationRun = async (data: EvaluationRunCreate) => {
  return fetchApi<EvaluationRunResponse>('/api/v1/evaluation/runs', {
    method: 'POST',
    body: JSON.stringify(data),
  });
};

export const getEvaluationRun = async (run_id: string) => {
  return fetchApi<EvaluationRunResponse>(`/api/v1/evaluation/runs/${run_id}`);
};

export const executeEvaluationRun = async (run_id: string) => {
  return fetchApi(`/api/v1/evaluation/runs/${run_id}/execute`, { method: 'POST' });
};

export const getEvaluationSummary = async () => {
  return fetchApi('/api/v1/evaluation/summary');
};

File: backend/packages/rag/schemas/evaluation.py
"""
Evaluation schemas - 评估请求/响应模型
"""
from typing import Optional, List, Dict, Any
from datetime import datetime
from pydantic import BaseModel, Field


class GoldenSampleCreate(BaseModel):
    """创建 Golden Sample"""
    kb_id: str = Field(..., description="知识库 ID")
    question: str = Field(..., description="问题")
    expected_answer: str = Field(..., description="期望答案")
    expected_context_ids: Optional[List[str]] = Field(None, description="期望的上下文 ID 列表")
    metadata: Optional[Dict[str, Any]] = Field(None, description="元数据")
    difficulty: Optional[str] = Field(None, description="难度", pattern="^(easy|medium|hard)$")
    category: Optional[str] = Field(None, description="分类")


class GoldenSampleResponse(BaseModel):
    """Golden Sample 响应"""
    id: str
    kb_id: str
    question: str
    expected_answer: str
    expected_context_ids: Optional[List[str]] = None
    metadata: Optional[Dict[str, Any]] = None

    class Config:
        from_attributes = True


class EvaluationRunCreate(BaseModel):
    """创建评估运行"""
    kb_id: str = Field(..., description="知识库 ID")
    name: str = Field(..., description="运行名称")
    evaluation_type: str = Field(..., description="评估类型")
    metrics: List[str] = Field(default_factory=list, description="评估指标列表")
    config: Optional[Dict[str, Any]] = Field(None, description="配置")


class EvaluationRunResponse(BaseModel):
    """评估运行响应"""
    id: str
    kb_id: str
    name: str
    evaluation_type: str
    metrics: List[str]
    config: Optional[Dict[str, Any]] = None
    status: str
    results: Optional[Dict[str, Any]] = None
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class EvaluationResult(BaseModel):
    """评估结果"""
    run_id: str
    total_samples: int
    evaluated: int
    avg_scores: Dict[str, float]
    overall_score: float
    details: List[Dict[str, Any]]
